docs/specification.md

```diff
@@ -1,34 +1,85 @@
-# Specification
+# Nushell Format Specification
 
-This specification serves as a starting point to document how `nufmt` should work when formatting nushell scripts. It's meant to be very basic at this point, gradually covering more of the language features.
+> rust has rsfmt
 
+> go has it too
 
-## Supported Features
+> formatting code is sweet
 
-### Indentation
+> and nushell needs one too
 
-There should be an `--indent` parameter to allow one to specify the number of spaces to use for indentation.
+These are some guidelines to format your [nushell scripts](https://github.com/nushell/nushell). It is not a must for the code to compile, but are well know practices in almost all programming languages.
 
-## Supported Commands
+This document also serves as a starting point to specify how `nufmt` should work when formatting nushell scripts. It's meant to be very basic at this point, gradually covering more of the language features.
 
-This is the list of the supported commands and their idiomatic formatting. Indentation will be covered by the `--indent` flag but for these examples, 2 spaces will be used.
+# Code layout:
 
-### if
+### Use 4 spaces to indent your code:
 
-```bash
-if condition {
-  # some thing
-} else if {
-  # some other thing
-} else {
-  # else the last thing
+```nu
+def one-function [] {
+    let number = 1
 }
 ```
 
-### for
+### Limit the characters per line
+
+There should be a reasonable limit for characters. 100 characters is a common ground between too much and too little
+
+```
+# transform this
+let resolve_this_really_long_line = ({|row| if $row.Type == DGRAM { $row | update Path { get I-Node } | update I-Node { get State } | update State "" } { $row } })
+
+# to this
+let resolve_this_really_long_line = (
+        { |row| if $row.Type == DGRAM {
+            $row | update Path { get I-Node } | update I-Node { get State } | update State "" }
+            { $row }
+        }
+    )
+```
 
-```bash
-for var in 0..100 {
-  # do something here
+### leave an empty line before writing a function
+
+```nu
+def first-function [] {
+    # do something
+}
+# leave an empty line between fn, like between second and third
+def second-function [] {
+    # do something
+}
+
+def third-function [] {
+    # do something
 }
-```
\ No newline at end of file
+```
+
+### Indent every `if`, `for` or `while` you have nested
+
+```
+while  {
+    for number in [1,2,3,4] {
+        if $number == 1 {
+            echo 'found 1!'
+        } else {
+            echo 'scanning'
+        }
+
+    }
+}
+```
+
+# If you opt-in `nufmt`, these are the supported features
+
+### Indentation
+
+There should be an `--indent` parameter to allow one to specify the number of spaces to use for indentation.
+
+## Supported Commands
+
+This is the list of the supported commands and their idiomatic formatting. Indentation will be covered by the `--indent` flag but for these examples, 4 spaces will be used.
+
```

Comment:
maybe add my original part back here instead of deleting the supported commands.